src/main/java/com/neilbaner/duke/Duke.java

```diff
@@ -0,0 +1,110 @@
+package com.neilbaner.duke;
+
+import com.neilbaner.duke.messages.Messages;
+import com.neilbaner.duke.task.*;
+
+import java.util.Scanner;
+
+public class Duke {
+    public static TaskList list = new TaskList();
+
+    public static void printTaskList() {
+        Messages.printHorizontalLine();
+        if (list.getSize() == 0) {
+            System.out.println("No tasks added yet");
+        }
+        for (Task task : list.getAllTasks()) {
+            System.out.println(task.toString());
+        }
+        Messages.printHorizontalLine();
+    }
+
+    public static void addToDo(String title) {
+        list.addTask(new ToDo(title));
+        Messages.printAddedToDo(title);
+    }
+
+    public static void addEvent(String title, String eventTime) {
+        list.addTask(new Event(title, eventTime));
+        Messages.printAddedEvent(title);
+    }
+
+    public static void addDeadline(String title, String dueDate) {
+        list.addTask(new Deadline(title, dueDate));
+        Messages.printAddedDeadline(title);
+    }
+
+    public static void markDone(int indexToMark) throws NumberFormatException, IndexOutOfBoundsException {
+        list.markTaskDone(indexToMark);
+        Messages.printMarkedDone(list.getTask(indexToMark).getTitle());
+    }
+
+    private static String getEventTime(String input) {
+        return input.substring(input.indexOf("/at") + 4).strip();
+    }
+
+    private static String getEventTitle(String input) {
+        return input.substring(6, input.indexOf("/at")).strip();
+    }
+
+    private static String getDeadlineDueDate(String input) {
+        return input.substring(input.indexOf("/by") + 4).strip();
+    }
+
+    private static int getIndexToMark(String input) {
+        return Integer.parseInt(input.substring(5)) - 1;
+    }
+
+    private static String getDeadlineTitle(String input) {
+        return input.substring(9, input.indexOf("/by")).strip();
+    }
+
+    private static boolean parseInput(String input) {
+        boolean toContinue = true;
+        if (input.toLowerCase().equals("bye")) {
```

Comment:
Consider creating String lowerCaseInput = input.toLowerCase() to reduce toLowerCase() method calls

Reply:
Good idea, thanks.